### ISSUE TITLE
Add config option for package format

### ISSUE THREAD

**Issue by jamesls** [aws/chalice#1424]
This avoids having to type `--pkg-format terraform|cloudformation` every time you want to deploy your code.

Originally requested in https://github.com/aws/chalice/issues/1423.

**Comment by devangmehta123** [comment]
@jamesls , yes, this is the right interpretation of what I meant. Even my other feature request, the one where I want to specify an S3 url for deployment.zip is part of this theme.

Just speculating a bit further down the road, `chalice deploy` as a single person option works very well. 

But to make chalice take off, there needs to be more polish added so that teams using infra as code have a simple and barrier free experience. I think serverless framework has done it in some way but it tries to compete against terraform or cloudfront, etc. rather than colloborate with it.

Chalice today is simple with its command line and no clutter experience. But it still takes a lot of discovery and tedium to make it work and we are left guessing as to the true level of future commitment to the team-based workflow.

If we could just take that barrier away......
